test-cases/tests/performance/k04-run-performance-test-against-rhoam.md

```diff
@@ -0,0 +1,98 @@
+
+## Description
+
+Run performance tests against 3scale + user SSO to validate the advertised load. Time estimation does not include the cluster provisioning.
+
+## Prerequisites
+
+- [oc CLI v4.3](https://docs.openshift.com/container-platform/3.6/cli_reference/get_started_cli.html#installing-the-cli)
+- [ocm CLI](https://github.com/openshift-online/ocm-cli/releases) installed locally
+- [jq v1.6](https://github.com/stedolan/jq/releases) installed locally
+- Python environment (python 3.x, pip, pipenv)
+- RHOAM cluster ready
+  - all the alerts should be green
+  - all the automated tests should pass
+
+## Steps
+
+1. Login via `oc` as a user with **cluster-admin** role (kubeadmin):
+
+   ```
+   oc login --token=<TOKEN> --server=https://api.<CLUSTER_NAME>.s1.devshift.org:6443
+   ```
+
+2. Make sure **nobody is using the cluster** for performing the test case so that performance test results are not affected by any unrelated workload.
+
+3. Create customer-like application using `customer-admin01` (or other user from dedicated-admin group)
+
+   ```bash
+   oc new-project httpbin
+   oc new-app jsmadis/httpbin
+   oc expose svc/httpbin
+   ```
+
+4. In terminal window #2, run the [alerts-during-perf-testing](https://github.com/integr8ly/integreatly-operator/blob/master/scripts/alerts-during-perf-testing.sh) script to capture alerts pending/firing during performance test run.
+
+5. Configure rate limiting to allow for enough requests per minute.
+
+- go to redhat-rhmi-operator namespace
+- see the `sku-limits-managed-api-service` Config Map
+- edit the value of `requests_per_unit`
+- wait for redeploy of ratelimit pods in redhat-rhmi-marin3r namespace
+
+  - should be done automatically in a few minutes
+
+    Note: This not possible for installations via addon-flow since Hive would revert your modifications to whatever is set in Managed Tenants repository in [sku-limits.yaml.j2](https://gitlab.cee.redhat.com/service/managed-tenants/-/blob/master/addons/managed-api-service/metadata/stage/sku-limits.yaml.j2) file.
+
+7. Run the performance test suite
+
+   The way to do it is described in [MGDAPI-238](https://issues.redhat.com/browse/MGDAPI-238) and in [Austin's doc](https://docs.google.com/document/d/1NJBUsieRkBLnN2PMAF5cpaH7uXq9mZCx1JQaT9Ruytk/edit?usp=sharing). Use [trepel branch](https://gitlab.cee.redhat.com/trepel/3scale-py-testsuite/-/tree/performance_tests). To validate the advertised load use [rhsso_tokens](https://gitlab.cee.redhat.com/trepel/3scale-py-testsuite/-/blob/performance_tests/testsuite/tests/performance/apicast/smoke/template_rhsso_tokens.hf.yaml) benchmark. It is set to have 10% of login flow requests. You will need to change `maxSessions` (~6000), `usersPerSec` (~25 to validate 20M load), `duration`, `maxDuration`, and [http.sharedConnections](https://gitlab.cee.redhat.com/trepel/3scale-py-testsuite/-/blob/performance_tests/testsuite/tests/performance/apicast/smoke/test_smoke_rhsso_tokens.py#L54-55) (~1200).
```

Comment:
```suggestion
   The way to do it is described in [MGDAPI-238](https://issues.redhat.com/browse/MGDAPI-238) and in [Austin's doc](https://docs.google.com/document/d/1NJBUsieRkBLnN2PMAF5cpaH7uXq9mZCx1JQaT9Ruytk/edit?usp=sharing). Use [trepel fork](https://gitlab.cee.redhat.com/trepel/3scale-py-testsuite/-/tree/performance_tests). To validate the advertised load use [rhsso_tokens](https://gitlab.cee.redhat.com/trepel/3scale-py-testsuite/-/blob/performance_tests/testsuite/tests/performance/apicast/smoke/template_rhsso_tokens.hf.yaml) benchmark. It is set to have 10% of login flow requests. You will need to change `maxSessions` (~6000), `usersPerSec` (~25 to validate 20M load), `duration`, `maxDuration`, and [http.sharedConnections](https://gitlab.cee.redhat.com/trepel/3scale-py-testsuite/-/blob/performance_tests/testsuite/tests/performance/apicast/smoke/test_smoke_rhsso_tokens.py#L54-55) (~1200).
```
change "trepel branch" to "trepel fork" :)

Reply:
ok, thanks. Will do